### PR TITLE
[WIP] Adding PAPI 7.0.0

### PR DESCRIPTION
**Pull request by fspiga**
New capabilities added, see https://icl.utk.edu/papi/news/news.html?id=384

Adapting the package recipe may require extra work. For now I am testing on Neoverse N1 and V1

### CODE REVIEW

**Comment by wyphan** [comment]
Just tried to build `papi@7.0.0 +cuda +rocm` on Zen2 running RHEL 8, it failed with the following error messages:
```console
==> Installing papi-7.0.0-cws3pj3pf4wjpef3pdpctvqszhnurkyg
==> No binary for papi-7.0.0-cws3pj3pf4wjpef3pdpctvqszhnurkyg found: installing from source
==> Fetching https://icl.cs.utk.edu/projects/papi/downloads/papi-7.0.0.tar.gz
==> No patches needed for papi
==> papi: Executing phase: 'autoreconf'
==> papi: Executing phase: 'configure'
==> papi: Executing phase: 'build'
==> Error: ProcessError: Command exited with status 2:
    'make' '-j8' 'V=1'

3 errors found in build log:
     1828    /tmp/cctWuYJy.o:(.bss+0x38): multiple definition of `cuDeviceGetNamePtr'
     1829    /tmp/cc737FJV.o:(.bss+0x300): first defined here
     1830    /tmp/cctWuYJy.o:(.bss+0x30): multiple definition of `cuDeviceGetCountPtr'
     1831    /tmp/cc737FJV.o:(.bss+0x308): first defined here
     1832    /tmp/cctWuYJy.o:(.bss+0x28): multiple definition of `cuDeviceGetAttributePtr'
     1833    /tmp/cc737FJV.o:(.bss+0x2c0): first defined here
  >> 1834    collect2: error: ld returned 1 exit status
  >> 1835    make[1]: *** [Makefile.inc:68: libpapi.so.7.0.0.0] Error 1
     1836    make[1]: Leaving directory '/scratch/wp11/spack-stage/spack-stage-papi-7.0.0-cws3pj3pf4wjpef3pdpctvqszhnurkyg/spack-sr
             c/src'
  >> 1837    make: *** [Rules.pfm4_pe:43: libpfm4/lib/libpfm.a] Error 2

See build log for details:
  /scratch/wp11/spack-stage/spack-stage-papi-7.0.0-cws3pj3pf4wjpef3pdpctvqszhnurkyg/spack-build-out.txt
```

Here's the full build log: [spack-build-out.txt](https://github.com/spack/spack/files/10044841/spack-build-out.txt)

Summoning @adanalis to help troubleshoot this.

Edit: in the meantime, I'm building `papi@7.0.0 +cuda` and `papi@7.0.0 +rocm` separately.

Update: `papi@7.0.0 +rocm` builds just fine, but `papi@7.0.0 +cuda` gives the exact same error as above.

**Comment by G-Ragghianti** [comment]
> 

Can you provide the full concretization?

**Comment by wyphan** [comment]
@G-Ragghianti sure, here it is:
```console
wp11@ufront:~/deploy$ spack/bin/spack -e spack-env spec papi@7.0.0 +cuda
Input spec
--------------------------------
papi@7.0.0+cuda

Concretized
--------------------------------
papi@7.0.0%gcc@11.3.0+cuda+example~infiniband~lmsensors~nvml~powercap~rapl~rocm~rocm_smi~sde+shared~static_tools build_system=autotools arch=linux-rhel8-zen2
    ^cuda@11.7%gcc@11.3.0~allow-unsupported-compilers~dev build_system=generic arch=linux-rhel8-zen2
```

**Comment by G-Ragghianti** [comment]
This should be resolved in papi upstream as it affects all users (not just in spack).

**Comment by G-Ragghianti** [comment]
I think also some more testing will be needed before papi 7 is ready for spack, so I think this PR is premature.  I usually add the new version to the spack package once I've finished all my tests, but I was not able to finish this process since papi tagged the 7.0.0 release.

**Comment by naromero77** [comment]
Can we try to resurrect this PR? And also add PAPI 7.0.1

**Comment by naromero77** [comment]
PAPI is a widely used package. I think it would be good to get this PR merged also included 7.0.1. 

Also, can the @G-Ragghianti please review.

**Comment by G-Ragghianti** [comment]
If version 7.0.0 is to be added, then version 7.0.1 also needs to be added because it includes an important fix that was delaying us from including version 7.0.0 initially.

File: var/spack/repos/builtin/packages/papi/package.py [hunk 1]
```diff
@@ -31,6 +31,7 @@ class Papi(AutotoolsPackage, ROCmPackage):
     git = "https://bitbucket.org/icl/papi/src/master/"
 
     version("master", branch="master")
+    version("7.0.0", sha256="799dcc317a47da96beaeed445812ecb6ae49df7c0504a4569797c1f7d74c4fd2")
```
Comment (alalazo) [inline var/spack/repos/builtin/packages/papi/package.py:34]: Can we add 7.0.1 directly (and remove 7.0.0), per @G-Ragghianti comment?

**Review by alalazo — CHANGES_REQUESTED** [review]

**Comment by naromero77** [comment]
@G-Ragghianti I may start a new PR because I have not heard back from the authors in a while. Can we add both 7.0.0 and 7.0.1?

**Comment by G-Ragghianti** [comment]
I'd say yes to the new PR.  I would not add 7.0.0 because it contains a bug that prevents it from working correctly in spack.

**Comment by naromero77** [comment]
Interesting... I remember being able to build 7.0.0 with Spack, but this was just the default variant.

**Comment by G-Ragghianti** [comment]
I don't remember the problem exactly, but I think it involved either the
"make test" or the smoke test.

**Comment by wyphan** [comment]
The `+cuda` variant also failed to build, like I [commented above](https://github.com/spack/spack/pull/33940#issuecomment-1320528155).

**Comment by wyphan** [comment]
@naromero77 I just minted the PR, please take a look.

**Comment by alalazo** [comment]
Closing, in favor of #38443

**Pull request closed without merging**